cmd/installer/cli/firewalld.go

```diff
@@ -0,0 +1,178 @@
+// service communication with default target ACCEPT, and opens the necessary ports in the default
+// zone for k0s and k8s components on the host network.
+func configureFirewalld(ctx context.Context, podNetwork, serviceNetwork string) error {
+	isActive, err := firewalld.IsFirewalldActive(ctx)
+	if err != nil {
+		return fmt.Errorf("check if firewalld is active: %w", err)
+	}
+	if !isActive {
+		logrus.Debugf("firewalld is not active, skipping configuration")
+		return nil
+	}
+
+	logrus.Debugf("firewalld is active, configuring")
+
+	cmdExists, err := firewalld.FirewallCmdExists(ctx)
+	if err != nil {
+		return fmt.Errorf("check if firewall-cmd exists: %w", err)
+	}
+	if !cmdExists {
+		logrus.Debugf("firewall-cmd not found but firewalld is active, skipping firewalld configuration")
+		return nil
+	}
+
+	err = ensureFirewalldECNetZone(ctx, podNetwork, serviceNetwork)
+	if err != nil {
+		return fmt.Errorf("ensure ec-net zone: %w", err)
+	}
+
+	err = ensureFirewalldDefaultZone(ctx)
+	if err != nil {
+		return fmt.Errorf("ensure default zone: %w", err)
+	}
+
+	err = firewalld.Reload(ctx)
+	if err != nil {
+		return fmt.Errorf("reload firewalld: %w", err)
+	}
+
+	return nil
+}
+
+// resetFirewalld removes all firewalld configuration added by the installer.
+func resetFirewalld(ctx context.Context) (finalErr error) {
+	cmdExists, err := firewalld.FirewallCmdExists(ctx)
+	if err != nil {
+		return fmt.Errorf("check if firewall-cmd exists: %w", err)
+	}
+	if !cmdExists {
+		return nil
+	}
+
+	err = resetFirewalldECNetZone(ctx)
+	if err != nil {
+		finalErr = multierr.Append(finalErr, fmt.Errorf("reset ec-net zone: %w", err))
+	}
+
+	err = resetFirewalldDefaultZone(ctx)
+	if err != nil {
+		finalErr = multierr.Append(finalErr, fmt.Errorf("reset default zone: %w", err))
+	}
+
+	err = firewalld.Reload(ctx)
+	if err != nil {
+		return fmt.Errorf("reload firewalld: %w", err)
+	}
+
+	return
+}
+
+func ensureFirewalldECNetZone(ctx context.Context, podNetwork, serviceNetwork string) error {
+	opts := []firewalld.Option{
+		firewalld.IsPermanent(),
+		firewalld.WithZone("ec-net"),
+	}
+
+	exists, err := firewalld.ZoneExists(ctx, "ec-net")
+	if err != nil {
+		return fmt.Errorf("check if ec-net zone exists: %w", err)
+	} else if !exists {
+		err = firewalld.NewZone(ctx, "ec-net", opts...)
+		if err != nil {
+			return fmt.Errorf("create ec-net zone: %w", err)
+		}
+	}
+
+	// Set the default target to ACCEPT for pod and service networks
+	err = firewalld.SetZoneTarget(ctx, "ACCEPT", opts...)
+	if err != nil {
+		return fmt.Errorf("set target to ACCEPT: %w", err)
+	}
+
+	err = firewalld.AddSourceToZone(ctx, podNetwork, opts...)
+	if err != nil {
+		return fmt.Errorf("add pod network source: %w", err)
+	}
+
+	err = firewalld.AddSourceToZone(ctx, serviceNetwork, opts...)
+	if err != nil {
+		return fmt.Errorf("add service network source: %w", err)
+	}
+
+	// Add the calico interfaces
+	// This is redundant and overlaps with the pod network but we add it anyway
```

Comment:
why though?

Reply:
well i cant see any harm in it. and because it is what was recommended on calico issues but preflights will not pass without adding sources since the interfaces do not exist when running preflights.